Content.Server/Physics/Controllers/PullController.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Collections.Generic;
+using Content.Shared.GameObjects.Components.Pulling;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Physics;
+using Robust.Shared.Physics.Controllers;
+
+namespace Content.Server.Physics.Controllers
+{
+    public class PullController : VirtualController
+    {
+        public override List<Type> UpdatesAfter => new() {typeof(MoverController)};
+
+        public override void UpdateBeforeSolve(bool prediction, float frameTime)
+        {
+            base.UpdateBeforeSolve(prediction, frameTime);
+
+            foreach (var pullable in ComponentManager.EntityQuery<SharedPullableComponent>())
+            {
+                if (pullable.MovingTo == null)
+                {
+                    continue;
+                }
+
+                if (!pullable.Owner.TryGetComponent<PhysicsComponent>(out var physics) ||
+                    physics.BodyType == BodyType.Static)
+                {
+                    pullable.MovingTo = null;
+                    continue;
+                }
+
+                var diff = pullable.MovingTo.Value.Position - pullable.Owner.Transform.Coordinates.Position;
+                physics.ApplyLinearImpulse(diff.Normalized * 20);
+            }
+        }
```

Comment:
Should ideally be event-based instead given there's a lot of pullable entities, like have sharedpullable just call a method directly.

Also using Coordinates as a diff is kinda stinky when it should be map position given if the parent's different it will go skewiff.

Also I would give it a minimum range check.

Reply:
Done.